FilmsApp/View/ListController/ListViewController.swift

```diff
@@ -48,28 +73,32 @@ extension ListViewController: UITableViewDelegate, UITableViewDataSource {
         let cell = tableView.dequeueReusableCell(withIdentifier: cell, for: indexPath)
         
         if let customCell = cell as? MovieListCell {
-        let text = data.films[indexPath.row].title
-            
-            if let url: URL? = data.films[indexPath.row].imageURL {
-            DispatchQueue.global().async { [weak self] in
-            if let data = try? Data(contentsOf: url!) {
-                if let image = UIImage(data: data) {
-                    DispatchQueue.main.async {
-                        customCell.movieListImage.image = image
-                        customCell.movieListLabel.text = text
-                    }
-                }
+            DispatchQueue.main.async {
+                customCell.updateAppearanceFor(content: self.data.films[indexPath.row], image: .none)
             }
-        }
-        }
             return customCell
         }
         return cell
     }
     
+    func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+        guard let cell = cell as? MovieListCell else { return }
+        
+        if let url: URL? = data.films[indexPath.row].imageURL {
+            DispatchQueue.main.async { [weak self] in
+                if let data = try? Data(contentsOf: url!) {
```

Comment:
Use "guard let" statement and combine creating of data and image in one statement. Also remove force unwrapping of a "url" property